### PR TITLE
139 small tests

### PR DESCRIPTION
**Pull request by cubap**
Starting to extract things from the overloaded db-controllers.js and organize the database methods and REST behavior more sensibly and testably.

### CODE REVIEW

File: config/index.js
Comment (cubap) [inline config/index.js:1]: This file prevents the need for `dotenv` across everything and makes config a simple import across the application.

File: bin/rerum_v1.js [hunk 1]
```diff
@@ -24,12 +25,10 @@ var server = http.createServer(app)
 const io = require('socket.io')(server)
 
 /**
- * Listen on provided port, on all network interfaces.
+ * Connect to Mongo and listen on provided port, on all network interfaces.
```
Comment (cubap) [inline bin/rerum_v1.js:28]: Changed to only listen when successfully connected to a database.

File: database/index.js
Comment (cubap) [inline database/index.js:1]: This file should contain an abstraction for the database, swappable if needed. The module exports a few things:

Export | Purpose | Note
--- | --- | ---
client | full MongoClient | Maybe shouldn't come out of this module.
connect | a simplified `connect(callback)` method for the app to launch with. | 
insert | save a single record in the RERUM way | either allow arrays or make `bulkInsert()` as well.
//delete | |
//update | |
//find | |

Things like `release` and `overwrite` are logical variations handled in the RESTful db-controller.

File: database/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * @param {Boolean} metadata.isExternalUpdate Document updates an externally referenced resource
+ * @returns URI of document to be saved. 
+ * @throws Error for bad data or slug and passes back any MongoDB errors
+ */
+async function insert(data, metadata = {}) {
+    if (!isObject(data)) throw new Error('Invalid data object')
+    if (!isValidURL(metadata.generator)) throw new Error('Invalid generator')
+    if (!ObjectID.isValid(metadata.slug)) throw new Error('Invalid slug')
+    const id = metadata.slug ?? new ObjectID().toHexString()
+    const configuredDocument = utils.configureRerumOptions(metadata.generator, Object.assign(data, { _id: id, "@id": `${config.id_prefix}${id}` }), false, metadata.isExternalUpdate)
```
Comment (cubap) [inline database/index.js:26]: utils.js can probably use some attention, but we're taking things one at a time.

File: globalConfig.json [hunk 3]
```diff
@@ -1 +1 @@
-{"mongoUri":"mongodb://127.0.0.1:56270/"}
\ No newline at end of file
+{"mongoUri":"mongodb://127.0.0.1:54979/"}
```
Comment (cubap) [inline globalConfig.json:1]: what even is this? dumb

**Comment by cubap** [comment]
@thehabes This work locally for you?

**Comment by thehabes** [comment]
Revisited via #181

**Pull request closed without merging**